instrumentation-annotations-support/src/main/java/io/opentelemetry/instrumentation/api/annotation/support/async/AsyncOperationEndStrategy.java

```diff
@@ -36,10 +37,35 @@ public interface AsyncOperationEndStrategy {
    * @return Either {@code asyncValue} or a value composing over {@code asyncValue} for notification
    *     of completion.
    */
-  <REQUEST, RESPONSE> Object end(
+  default <REQUEST, RESPONSE> Object end(
       Instrumenter<REQUEST, RESPONSE> instrumenter,
       Context context,
       REQUEST request,
       Object asyncValue,
+      Class<RESPONSE> responseType) {
+    return end(instrumenter::end, context, request, asyncValue, responseType);
+  }
+
+  /**
+   * Composes over {@code asyncValue} and delays the {@link AsyncOperationEndHandler#handle(Context,
```

Comment:
@brunobat had a good question about the name of this new interface, how about

```suggestion
   * Composes over {@code asyncValue} and delays the {@link AsyncOperationCallback#onEnd(Context,
```

Reply:
I would prefer `AsyncEndHandler` or `AsyncCallback` too as you mentioned. But I think we will do it in another PR because it's out of this PR's scope. Currently I only change it to `AsyncOperationCallback#onEnd`.

Reply:
sure!